system/jlib/jptree.cpp

```diff
@@ -7961,3 +7961,262 @@ jlib_decl IPropertyTree * loadConfiguration(const char * defaultYaml, const char
+        noRoot = 0 != ((unsigned)readerOptions & (unsigned)ptr_noRoot);
+        if (!noRoot)
+            rootTag.set("_object_"); //may support _array_ option later
+    }
+    ~CYAMLBufferReader()
+    {
+        yaml_parser_delete(&parser);
+    }
+
+    yaml_event_type_t nextEvent(yaml_event_t &event, yaml_event_type_t final=YAML_NO_EVENT, yaml_event_type_t expected=YAML_NO_EVENT, const char *error="")
+    {
+        if (!yaml_parser_parse(&parser, &event))
+            throw makeStringExceptionV(99, "libyaml parser error %s", parser.problem);
+        if (event.type!=final && expected!=YAML_NO_EVENT && event.type!=expected)
+            throw makeStringExceptionV(99, "libyaml parser %s", error);
+        return event.type;
+    }
+
+    virtual void loadSequence(const char *tagname)
+    {
+        if (!tagname || !*tagname) //if unmapped (unnamed) sequences are possible have to decide how to name them in the ptree, later
+            throw makeStringException(99, "libyaml parser expected sequence name");
+
+        yaml_event_t event;
+        yaml_event_type_t eventType = YAML_NO_EVENT;
+
+        while (eventType!=YAML_SEQUENCE_END_EVENT)
+        {
+            eventType = nextEvent(event);
+
+            switch (eventType)
+            {
+            case YAML_MAPPING_START_EVENT: //child map
+                loadMap(tagname);
+                break;
+            case YAML_SEQUENCE_START_EVENT:
+                //todo
+                break;
+            case YAML_SCALAR_EVENT:
+                iEvent->beginNode(tagname, parser.offset);
+                iEvent->endNode(tagname, event.data.scalar.length, (const void *)event.data.scalar.value, false, parser.offset);
+                break;
+            case YAML_ALIAS_EVENT: //reference to an anchor, ignore for now
+                iEvent->beginNode(tagname, parser.offset);
+                iEvent->endNode(tagname, 0, nullptr, false, parser.offset);
+                break;
+            case YAML_SEQUENCE_END_EVENT: //done
+                break;
+            case YAML_NO_EVENT:
+            case YAML_MAPPING_END_EVENT:
+            case YAML_STREAM_START_EVENT:
+            case YAML_STREAM_END_EVENT:
+            case YAML_DOCUMENT_START_EVENT:
+            case YAML_DOCUMENT_END_EVENT:
+            default:
+                //shouldn't be here
+                break;
+            }
+
+            yaml_event_delete(&event);
+        }
+    }
+    virtual void loadMap(const char *tagname)
+    {
+        StringBuffer content;
+        if (tagname && *tagname)
+            iEvent->beginNode(tagname, parser.offset);
+
+        yaml_event_t event;
+        yaml_event_type_t eventType = YAML_NO_EVENT;
+
+        while (eventType!=YAML_MAPPING_END_EVENT)
+        {
+            eventType = nextEvent(event, YAML_MAPPING_END_EVENT, YAML_SCALAR_EVENT, "expected map to start with scalar name");
+            if (eventType==YAML_MAPPING_END_EVENT)
+                continue;
```

Comment:
minor: Should this also call yaml_event_delete(&event)?  I suspect it will have no effect,

Reply:
Agreed.  No leaks detected but could change.

Reply:
actually value does seem to be null terminated, so should be able to use directly.